### PR TITLE
Stripe

### PR DESCRIPTION
**Pull request by lemieux**
Let me know if I forgot some!

@alavers @mspensieri

### CODE REVIEW

File: src/api/stripe.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    const url = this.getFullURL('appUsers', userId, 'stripe', 'customer');
+    return this.validateAuthHeaders().then((headers) => {
+      return http('POST', url, {
+        token
+      }, headers);
+    });
+  }
+
+  checkout(userId, actionId, token) {
```
Comment (mspensieri) [inline src/api/stripe.js]: `/checkout` isn't supposed to be part of the public API, it's used exclusively by the SMS checkout page

Comment (lemieux) [inline src/api/stripe.js]: gone

File: tests/specs/api/stripe.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  beforeEach(() => {
+    httpSpy = httpMock.mock();
+    api = new StripeApi(serviceUrl, httpHeaders);
+  });
+
+  afterEach(() => {
+    httpMock.restore();
+  });
+
+  describe('#updateCustomer', () => {
```
Comment (mspensieri) [inline tests/specs/api/stripe.spec.js]: `POST /customer` is a jwt-only API, but your test is using `appToken`. Lib should probably return an error if app token is the only provided auth method when you attempt to hit this endpoint

File: src/api/stripe.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @class StripeApi
+ * @extends BaseApi
+ */
+export class StripeApi extends BaseApi {
+
+  updateCustomer(userId, token) {
+    if (!token) {
+      return Promise.reject(new Error('Must a Stripe token.'))
```
Comment (mspensieri) [inline src/api/stripe.js]: Must _provide_ a Stripe token

File: src/api/stripe.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    return this.validateAuthHeaders().then((headers) => {
+      return http('POST', url, {
+        token
+      }, headers);
+    });
+  }
+
+  createTransaction(userId, actionId, token) {
+    if (!actionId) {
+      return Promise.reject(new Error('Must an action id.'));
```
Comment (mspensieri) [inline src/api/stripe.js]: _provide_

File: src/api/stripe.js (hunk 3 above)
Comment (lemieux) [inline src/api/stripe.js]: errr ... select and delete went wrong..

File: tests/specs/api/stripe.spec.js (hunk 2 above)
Comment (lemieux) [inline tests/specs/api/stripe.spec.js]: fixed.

**Comment by mspensieri** [comment]
:+1:

**Pull request merged**